client/state/data-layer/wpcom/read/streams/index.js

```diff
@@ -208,17 +206,13 @@ const streamApis = {
 	discover: {
 		path: ( { streamKey } ) => {
 			if ( streamKeySuffix( streamKey ).includes( 'recommended' ) ) {
-				if ( config.isEnabled( 'reader/discover-stream' ) ) {
-					return '/read/streams/discover';
-				}
-
-				return '/read/tags/cards';
+				return '/read/streams/discover';
 			} else if ( streamKeySuffix( streamKey ).includes( 'latest' ) ) {
 				return '/read/tags/posts';
 			} else if ( streamKeySuffix( streamKey ).includes( 'firstposts' ) ) {
 				return '/read/streams/first-posts';
 			}
-			return `/read/tags/${ streamKeySuffix( streamKey ) }/cards`;
+			return `/read/streams/discover?tags=${ streamKeySuffix( streamKey ) }`;
```

Comment:
Just to confirm: I gather that we don't need any extra encoding when we're moving the stream key to a query arg?

Reply:
No, it shouldn't be necessary. The Discover page is already using the user's locale and that gets automatically picked up by the endpoint.